apps/authenticator/lib/sign_in/commands/resource_owner.ex

```diff
@@ -122,6 +122,10 @@ defmodule Authenticator.SignIn.Commands.ResourceOwner do
     end
   end
 
+  defp valid_totp?(%{totp: nil}, nil), do: true
+  defp valid_totp?(%{totp: nil}, code) when is_binary(code), do: false
+  defp valid_totp?(%{totp: totp}, code) when is_binary(code), do: Port.valid_totp?(totp, code)
```

Comment:
```suggestion
  defp valid_totp?(%{totp: nil}, <<code::bitstring>>), do: false
  defp valid_totp?(%{totp: totp}, <<code::bitstring>>), do: Port.valid_totp?(totp, code)
```

Reply:
I think guards are cleaner in these cases.